source/common/network/dns_impl.cc

```diff
@@ -115,17 +131,26 @@ void DnsResolverImpl::onAresSocketStateChange(int fd, int read, int write) {
                           (write ? Event::FileReadyType::Write : 0));
 }
 
-ActiveDnsQuery* DnsResolverImpl::resolve(const std::string& dns_name, ResolveCb callback) {
+ActiveDnsQuery* DnsResolverImpl::resolve(const std::string& dns_name,
+                                         const DnsLookupFamily& dns_lookup_family,
+                                         ResolveCb callback) {
   std::unique_ptr<PendingResolution> pending_resolution(new PendingResolution());
   pending_resolution->callback_ = callback;
+  pending_resolution->channel_ = channel_;
```

Comment:
As a cleanup to my earlier mess, you could make all of these constructor args and initialize them in the constructor's member initializor list, allowing them to be made `const`.